### ISSUE TITLE
Add a nice way to define constants

### ISSUE THREAD

**Issue by eXon** [andywer/webpack-blocks#17]
It would be great to have at the core level of webpack-blocks a way to define constants within the code. Right now, we could do that using the DefinePlugin but I think it deserves a nicer way.

Here is my proposal:

```js
const { createConfig, defineConstant } = require('@webpack-blocks/webpack')

module.exports = createConfig([
  defineConstant('process.env.NODE_ENV', process.env.NODE_ENV),
  defineConstant('__API__URL', process.env.API_URL),
  // ...
])
```

And it would be transformed by:

```js
new webpack.DefinePlugin({
  'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
  '__API_URL__', JSON.stringify(process.env.API_URL)
})
```

We might want to stack them instead of doing it with a single plugin. I don't know if it will be possible the way things are built with webpack-blocks.

Or maybe it could be a defineConstants that takes more than a single one.

What do you think? I can make a PR if you want to experiment with it.

**Comment by andywer** [comment]
I also thought about the DefinePlugin, usability and edge cases, but without a clear answer so far.

Exporting a `defineConstant` helper function is basically no problem, but I am not sure if it really adds any value (except for having an easier-to-understand name).

What I am wondering right now is if you could have multiple instances of the DefinePlugin in your config or if that crashes or introduces noticeable performance penalties.

**Comment by andywer** [comment]
@eXon When you wrote "want to stack them" and "don't know if it will be possible the way things are built with webpack-blocks" I guess you mean if it's possible to have multiple `defineConstant()` but get a single `new webpack.DefinePlugin` in the end?

**Comment by eXon** [comment]
It is what I meant by stacking them. You can use multiple `DefinePlugin` but I don't know what's the performance impact. It is probably faster to have a single one.

Maybe the plugin could be added only when you call `createConfg` (haven't checked the implementation details yet).

That way, it is easier to override a value when you share configs between a client and server webpack config or if you want to share a common config between multiple projects. 

This is mainly syntax sugar but there is a few benefits when sharing configs.

**Comment by andywer** [comment]
True. Being able to have multiple `defineConstants()` might be a real benefit in larger configs.

It is not achievable very easily right now without increasing code complexity under webpack-block's hood, I think. But there should be some way and it should be easy and elegant. Got to think about that...

**Comment by eXon** [comment]
@andywer Here is an idea how it could be achieved without adding too much complexity:

After invoking [invokeConfigSetters](https://github.com/andywer/webpack-blocks/blob/master/packages/core/src/index.js#L26), we could let webpack-blocks do custom post-processing on the config.

That way, you can save your constants within an unused webpack config and the post-processing could transform that correctly.

The blocks would need a subscription system for the post-processing but I'm not sure how we can achieve this automatically.

**Comment by andywer** [comment]
Yep. Also thought about pre and post processing hooks on blocks 👍

Not sure if it's a good idea to put it somewhere in the webpack config object, though.

This screams for at least one good night's sleep, a separate issue and a feature branch to play with the idea! :)

**Comment by eXon** [comment]
In webpacker I was using a state that could contain anything before creating the webpack config. Maybe you could provide a state object that will be given to the post-processing. I'll try to play with it when I have sometime. 

I though about pre-processing too but I fail to think about a single use-case. 

BTW is there a special way to setup a symlink for using the @webpack-blocks with my local version? I cloned the repo but I can't setup it up correctly so that require('@webpack-blocks/core') point to my local version.

**Comment by andywer** [comment]
`lerna bootstrap` should npm-link everything. Alternatively: Go into the package and `npm link`, go to where you want to use it and `npm link @webpack-blocks/xyz`.

The pre-processing hook may be handy to add another mime type matcher (there is that mime type - regex abstraction).

**Comment by andywer** [comment]
Side note: Blocked by #25. This constant-definition issue then again pretty much blocks #24.
